Added TextInput and Response Functionality

ojasvib opened the pull request.

ojasvib committed:
Added TextInput and Response Functionality

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/MainActivity.kt:
```diff
@@ -50,54 +25,17 @@ class MainActivity : ComponentActivity() {
     Scaffold(
-        topBar = { TopAppBar(title = { Text(text = "Memify")})},
-        content = {ImagePicker()},
+        content = { PhotoReasoningRoute() },
```
Should be `PhotoReasoningScreen`

sanskar10100 replied:
Pass a modifier to the composable with the inner padding provided by the scaffold

sanskar10100 replied:
Otherwise remove Scaffold, directly call `PhotoReasoningScreen`, and add `Modifier.navigationBarPadding()`

ojasvib replied:
Replaced the Scaffold with a box.

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -0,0 +1,244 @@
+import kotlinx.coroutines.launch
+
+@Composable
+internal fun PhotoReasoningRoute(
```
Remove `internal`

sanskar10100 replied:
Rename to `PhotoReasoningScreen`

ojasvib replied:
Done

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -0,0 +1,244 @@
+
+    PhotoReasoningScreen(
+        uiState = photoReasoningUiState,
+        onReasonClicked = { inputText, selectedImage ->
+            coroutineScope.launch {
+                val bitmap = try {
+                    val imageReq = imageRequestBuilder.data(selectedImage)
+                        .size(768)
+                        .precision(Precision.EXACT)
+                        .build()
+
+                    val result = imageLoader.execute(imageReq)
+                    if (result is SuccessResult) {
+                        (result.drawable as BitmapDrawable).bitmap
+                    } else {
+                        null
+                    }
+                } catch (e: Exception) {
+                    null
+                }
+
+                // Call reason function with inputText and bitmap
+                if (bitmap != null) {
+                    viewModel.reason(inputText, bitmap)
+                }
```
This should happen in ViewModel

ojasvib replied:
Done by @sanskar10100

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/PhotoReasoningViewModel.kt:
```diff
@@ -0,0 +1,57 @@
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class PhotoReasoningViewModel : ViewModel() {
```
If need context, use `AndroidViewModel`

ojasvib replied:
Done by @sanskar10100

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -0,0 +1,244 @@
+            Row(
+                modifier = Modifier.padding(top = 16.dp)
+            ) {
+                IconButton(
+                    onClick = {
+                        pickMedia.launch(
+                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
+                        )
+                    },
+                    modifier = Modifier
+                        .padding(all = 4.dp)
+                        .align(Alignment.CenterVertically)
+                ) {
+                    Icon(
+                        Icons.Rounded.Add,
+                        contentDescription = stringResource(R.string.add_image),
+                    )
+                }
```
The modifier and clickable can be applied directly to the Icon. `IconButton` is not required.

ojasvib replied:
Done

sanskar10100 commented on app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt:
```diff
@@ -0,0 +1,244 @@
+                    modifier = Modifier
+                        .fillMaxWidth(0.8f)
+                )
+                TextButton(
+                    onClick = {
+                        if (userQuestion.isNotBlank()) {
+                            imageUri?.let { onReasonClicked(userQuestion, it) }
+                        }
+                    },
+                    modifier = Modifier
+                        .padding(all = 4.dp)
+                        .align(Alignment.CenterVertically)
+                ) {
+                    Text(stringResource(R.string.action_go))
+                }
+            }
```
Add clickable to text, no need to use TextButton

ojasvib replied:
Done

sanskar10100 committed:
move bitmap conversion to viewmodel

ojasvib committed:
Refactor UI components in PhotoReasoningScreen and MainScreen

- Rename PhotoReasoningRoute to PhotoReasoningScreen for clarity.
- Remove redundant internal modifier from PhotoReasoningScreen.
- Fix inner padding handling in Scaffold in MainScreen by passing it to PhotoReasoningScreen.
- Simplify clickable behavior by removing unnecessary IconButton and applying clickable directly to Icon.
- Refactor TextButton usage by applying clickable modifier directly to Text for a cleaner UI structure.
